README.md

```diff
@@ -555,6 +548,58 @@ class MyConsumer < Deimos::ActiveRecordConsumer
   def record_attributes(payload)
     super.merge(:some_field => 'some_value')
   end
+
+  # Optional override to change the attributes used for identifying records
+  def record_key(payload)
+    super
+  end
+end
+```
+
+#### Batch Consumers
+
+Deimos also provides a batch consumption mode for `ActiveRecordConsumer` which
+processes groups of messages at once using the ActiveRecord backend. 
+
+Batch ActiveRecord consumers make use of the
+[activerecord-import](https://github.com/zdennis/activerecord-import) to insert
+or update multiple records in bulk SQL statements. This reduces processing
+time at the cost of skipping ActiveRecord callbacks for individual records.
+Deleted records (tombstones) are grouped into `delete_all` calls and thus also
+skip `destroy` callbacks.
+
+Batch consumption is used when the `delivery` setting for your consumer is set to `inline_batch`.
+
+**Note**: Currently, batch consumption only supports only primary keys as identifiers out of the box. See
+[the specs](spec/active_record_batch_consumer_spec.rb) for an example of how to use compound keys.
+
+A sample batch consumer would look as follows:
+
+```ruby
+class MyConsumer < Deimos::ActiveRecordConsumer
+  schema 'MySchema'
+  key_config field: 'my_field'
+  record_class Widget
+
+  # Controls whether the batch is compacted before consuming.
+  # If true, only the last message for each unique key in a batch will be
+  # processed.
+  # If false, each message will be processed in order.
+  compacted false
```

Comment:
Should this also be added to the configuration side?

Reply:
Would this be in the `consumer` `setting_object`? I considered this but I wasn't sure if it made sense there as it only applied to ActiveRecord batch consumers. I can move it there if you think it belongs, but I figured it was very similar to the `record_class` config

Reply:
Hmm... yeah, OK. Let's leave it here for now. If we start accumulating more of these we might want to add them in.